### PR TITLE
[improvement](fixreplica) move to healthy replica when fix replica bad

### PR DESCRIPTION
**Pull request by yiguolei**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by yiguolei** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/planner/OlapScanNode.java [hunk 1]
```diff
@@ -774,8 +774,21 @@ private void addScanRangeLocations(Partition partition,
...
                 Replica replica = replicas.get(useFixReplica >= replicas.size() ? replicas.size() - 1 : useFixReplica);
-                replicas.clear();
-                replicas.add(replica);
+                if (ConnectContext.get().getSessionVariable().fallbackOtherReplicaWhenFixedCorrupt) {
+                    Backend backend = Env.getCurrentSystemInfo().getBackend(replica.getBackendId());
+                    // If the fixed replica is bad, then not clear the replicas using random replica
+                    if (backend == null || !backend.isAlive()) {
+                        LOG.debug("backend {} not exists or is not alive for replica {}", replica.getBackendId(),
+                                replica.getId());
+                        Collections.shuffle(replicas);
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/planner/OlapScanNode.java:783]: Should we remove the bad replica before shuffle?

**Review by morningman — APPROVED** [review]
LGTM

**Review by xinyiZzz — APPROVED** [review]
LGTM

**Pull request merged**